sdk/eventhub/Azure.Messaging.EventHubs/MIGRATIONGUIDE.md

```diff
@@ -0,0 +1,393 @@
+        // the cancellation token.
+    }
+}
+```
+
+See [`ReadEvents.cs`](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample09_ReadEventsFromAKnownPosition.cs) for a sample program demonstrating this.
+
+### Migrating code from `EventProcessorHost` to `EventProcessorClient` for reading events
+
+In v4, `EventProcessorHost` allowed you to balance the load between multiple instances of your program and checkpoint
+events when receiving. Developers would have to create and register a concrete implementation of `IEventProcessor` to
+begin consuming events.
+
+In v5, `EventProcessorClient` allows you to do the same, the development model is made simpler by registering functions that would be invoked for each event. To use
+this, include [Azure.Messaging.EventHubs.Processor](https://www.nuget.org/packages/Azure.Messaging.EventHubs.Processor) as a dependency.
+
+The following code in v4:
+
+```csharp
+private static void Main(String[] args) {
+    EventProcessorHost eventProcessorHost = new EventProcessorHost("my-eventhub-name", "my-consumer-group", "connection-string-for-an-event-hub", "storage-connection-string", "storage-container-name");
+    
+    // Registers the Event Processor Host and starts receiving messages
+    await eventProcessorHost.RegisterEventProcessorAsync<SimpleEventProcessor>();
+    // When you are finished processing events.
+    await eventProcessorHost.UnregisterEventProcessorAsync();
+}
+
+public class SimpleEventProcessor implements IEventProcessor {
+    public Task CloseAsync(PartitionContext context, CloseReason reason)
+    {
+         Console.WriteLine($"Processor Shutting Down. Partition '{context.PartitionId}', Reason: '{reason}'.");
+         return Task.CompletedTask;
+    }
+    public Task OpenAsync(PartitionContext context)
+    {
+        Console.WriteLine($"SimpleEventProcessor initialized. Partition: '{context.PartitionId}'");
+        return Task.CompletedTask;
+    }
+    public Task ProcessErrorAsync(PartitionContext context, Exception error)
+    {
+        Console.WriteLine($"Error on Partition: {context.PartitionId}, Error: {error.Message}");
+        return Task.CompletedTask;
+    }
+    public Task ProcessEventsAsync(PartitionContext context, IEnumerable<EventData> messages)
+    {
+        foreach (var eventData in messages)
+        {
+            var data = Encoding.UTF8.GetString(eventData.Body.Array, eventData.Body.Offset, eventData.Body.Count);
+            Console.WriteLine($"Message received. Partition: '{context.PartitionId}', Data: '{data}', Partition Key: '{eventData.SystemProperties.PartitionKey}'");
+        }
+
+        return context.CheckpointAsync();
+    }
+}
+```
+
+And in v5, in order to use the `EventProcessorClient`, handlers for event processing and errors must be provided.  These handlers are considered self-contained and developers are responsible for ensuring that exceptions within the handler code are accounted for.
```

Comment:
To keep the snippets equivalent, the handlers in the `v5` snippet should behave the same way the handlers in the `v4` one do, simply printing the same messages and returning a `Task.CompletedTask`. We need to illustrate how the `OpenAsync` and `CloseAsync` methods are converted into the `PartitionInitializingAsync` and `PartitionClosingAsync` handlers.